pkg/pipeline/templates/terraformed.go.tmpl

```diff
@@ -38,6 +38,16 @@ func (tr *{{ .CRD.Kind }}) GetObservation() (map[string]interface{}, error) {
 
 // SetObservation for this {{ .CRD.Kind }}
 func (tr *{{ .CRD.Kind }}) SetObservation(obs map[string]interface{}) error {
+    id, ok := obs["id"]
+    if !ok {
+        return errors.New("attribute id not found in observed state")
+    }
+    idStr, ok := id.(string)
+    if !ok {
+        return errors.New("attribute id in observed state is not of type string")
+    }
+    tr.Status.AtProvider.ID = idStr
```

Comment:
What do you think about having this under status directly, i.e. `tr.Status.ID`? I'm worried that we'll see many conflicts in the observation since `id` is a common field name.

Reply:
Maybe another name, like `TerrajetID` or `JetID`?

Reply:
As we have discussed in the sync, my observation is that the `id` is in the set of returned attributes for the Terraform state, and hence I don't expect a Terraform resource schema to define a conflicting attribute with it. As an example:
```hcl
 "resources": [
    {
      "mode": "managed",
      "type": "azurerm_resource_group",
      "name": "example",
      "provider": "provider[\"registry.terraform.io/hashicorp/azurerm\"]",
      "instances": [
        {
          "schema_version": 0,
          "attributes": {
            "id": "/subscriptions/.../resourceGroups/example-rg-alper",
            "location": "eastus",
            "name": "example-rg-alper",
            "tags": null,
            "timeouts": null
          },
...
```

And because this is the persistent ID of the resource as observed on the Cloud provider side, I had decided to put this into `status.atProvider`. But I had also considered making it directly available at `status`. I'm also fine with it. Just semantically I think it makes sense at `status.atProvider`.